### PR TITLE
Add Environment Variables page

### PR DESCRIPTION
**Pull request by S3Prototype**
This PR:

- Adds a dedicated clerk environment variables page
- Links to this page wherever the Clerk component props related to redirects are mentioned
  - The exception is `redirectUrl`, which has no corresponding env var
- Update the nextjs core 2 migration guide. After [a chat with Bryce](https://clerkinc.slack.com/archives/C06A9NUVC6L/p1712598701164999), I realized it had incorrect info about which env var to use for `CLERK_JS_URL`

[Preview](https://docs-preview-885.clerkpreview.com/docs/deployments/clerk-environment-variables)

### CODE REVIEW

File: docs/references/javascript/clerk/handle-navigation.mdx [hunk 1]
```diff
@@ -79,8 +79,8 @@ function handleRedirectCallback(
 
 | Name | Type | Description |
 | --- | --- | --- |
-| `afterSignUpUrl` | `string \| undefined \| null` | Full URL or path to navigate after successful sign up. Defaults to `/`. |
-| `afterSignInUrl` | `string \| undefined \| null` | Full URL or path to navigate after successful sign in. Defaults to `/`. |
+| `afterSignUpUrl` | `string \| undefined \| null` | Full URL or path to navigate after successful sign up. Defaults to `/`. Use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
+| `afterSignInUrl` | `string \| undefined \| null` | Full URL or path to navigate after successful sign in. Defaults to `/`. Use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
```
Comment (alexisintech) [inline docs/references/javascript/clerk/handle-navigation.mdx:83]:
```suggestion
| `afterSignUpUrl` | `string \| undefined \| null` | Full URL or path to navigate after successful sign up. Defaults to `/`. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
| `afterSignInUrl` | `string \| undefined \| null` | Full URL or path to navigate after successful sign in. Defaults to `/`. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
```

File: docs/references/javascript/types/sign-in-redirect-options.mdx [hunk 2]
```diff
@@ -11,7 +11,7 @@ An interface that provides options for sign-in redirect methods.
 
 | Name | Type | Description |
 | --- | --- | --- |
-| `afterSignInUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-in. Defaults to `/`. |
-| `afterSignUpUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-up. Defaults to `/`. |
+| `afterSignInUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-in. Defaults to `/`. Use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
+| `afterSignUpUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-up. Defaults to `/`. Use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
```
Comment (alexisintech) [inline docs/references/javascript/types/sign-in-redirect-options.mdx:15]:
```suggestion
| `afterSignInUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-in. Defaults to `/`. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
| `afterSignUpUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-up. Defaults to `/`. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
```

File: docs/references/javascript/types/sign-up-redirect-options.mdx [hunk 3]
```diff
@@ -11,7 +11,7 @@ An interface that provides options for sign-up redirect methods.
 
 | Name | Type | Description |
 | --- | --- | --- |
-| `afterSignInUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-in. Defaults to `/`. |
-| `afterSignUpUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-up. Defaults to `/`. |
+| `afterSignInUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-in. Defaults to `/`. Use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
+| `afterSignUpUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-up. Defaults to `/`. Use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
```
Comment (alexisintech) [inline docs/references/javascript/types/sign-up-redirect-options.mdx:15]:
```suggestion
| `afterSignInUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-in. Defaults to `/`. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
| `afterSignUpUrl?` | `string \| null \| undefined` | Full URL or path to navigate after successful sign-up. Defaults to `/`. It's recommended to use [the environment variable](/docs/deployments/clerk-environment-variables#sign-in-and-sign-up-redirects) instead. |
```

File: docs/references/nextjs/current-user.mdx
Comment (alexisintech) [inline docs/references/nextjs/current-user.mdx:1]: this page should not be getting reverted

File: docs/testing/test-emails-and-phones.mdx
Comment (alexisintech) [inline docs/testing/test-emails-and-phones.mdx:1]: this page should not be getting reverted.

**Review by alexisintech — CHANGES_REQUESTED** [review]

File: docs/references/backend/overview.mdx
[the remark this answers is not in the document]
Comment (S3Prototype) [reply docs/references/backend/overview.mdx:1]: rebase error

**Pull request merged**